lib/common/grpc-service.js

```diff
@@ -457,4 +458,37 @@ GrpcService.prototype.getGrpcCredentials_ = function(callback) {
   });
 };
 
+/**
+ * @private
+ */
+GrpcService.prototype.loadProtoFile_ = function(protoConfig, config) {
+  var rootDir = googleProtoFiles('..')
+
+  var grpcOpts = {
+    binaryAsBase64: true,
+    convertFieldsToCamelCase: true
+  };
+
+  if (is.string(protoConfig)) {
+    protoConfig = {
+      path: protoConfig
+    };
+  }
+
+  var services = grpc.load({
+    root: rootDir,
+    file: path.relative(rootDir, protoConfig.path)
+  }, 'proto', grpcOpts);
+
+  var serviceName = protoConfig.service || this.service || config.service;
```

Comment:
I'm still pretty new to this code, but it seems like `config.service` should beat `this.service` since it was explicitly provided.